pkg/routeagent-driver/handlers/kubeproxy-iptables/kp-iptables.go

```diff
@@ -0,0 +1,67 @@
+package kp_iptables
+
+import (
+	"fmt"
+	"os"
+
+	"k8s.io/klog"
+
+	"github.com/submariner-io/submariner/pkg/event"
+	cni_driver "github.com/submariner-io/submariner/pkg/routeagent-driver/cni-driver"
+)
+
+type SyncHandler struct {
+	event.HandlerBase
+	clusterID        string
+	namespace        string
+	localClusterCidr []string
+	localServiceCidr []string
+
+	hostname string
+	cniIface *cni_driver.CniInterface
+}
+
+func NewSyncHandler(clusterID, namespace string, clusterCidr, serviceCidr []string) *SyncHandler {
+	return &SyncHandler{
+		clusterID:        clusterID,
+		namespace:        namespace,
+		localClusterCidr: clusterCidr,
+		localServiceCidr: serviceCidr,
+	}
+}
+
+func (kp *SyncHandler) GetName() string {
+	return "kubeproxy-iptables-handler"
+}
+
+func (kp *SyncHandler) GetNetworkPlugin() string {
+	return event.AnyNetworkPlugin
+}
+
+func (kp *SyncHandler) Init() error {
+	var err error
+	kp.hostname, err = os.Hostname()
+	if err != nil {
+		klog.Fatalf("unable to determine hostname: %v", err)
+	}
+
+	cniIface, err := cni_driver.DiscoverCNIInterface(kp.localClusterCidr[0])
+	if err == nil {
+		// Configure CNI Specific changes
+		kp.cniIface = cniIface
+		err := cni_driver.ToggleCNISpecificConfiguration(kp.cniIface.Name)
+		if err != nil {
+			return fmt.Errorf("ToggleCNISpecificConfiguration returned error. %v", err)
+		}
+	} else {
+		klog.Errorf("DiscoverCNIInterface returned error %v", err)
+	}
```

Comment:
I would make this block a whole function we can call.

cni_interface.Setup()

Reply:
I can do that, but we also need an API that returns the CNIInterface on the host.